src/intercomparison.py

```diff
@@ -0,0 +1,67 @@
+import sklearn.metrics
+import geopandas as gpd
+import pandas as pd
+import rasterio as rio
+from pyproj import Proj
+from pyproj import transform
+import numpy as np
+import json
+import os
+
+def run_comparison(validationPath: str, cropmapPath: str, validation_projection = None, cropmap_projection: str = None, asJSON = False, subset: str = None):
+    
+  # Assume datatypes to be csv and tiff for validation and cropmap respectively
+  report = []
+  cropmap = rio.open(cropmapPath)
+  validation = gpd.read_file(validationPath)
+
+  # Resets csv to specified subset
+  if subset != None:
+    validation = validation[validation['subset'] == subset]
+
+  validation['crop_probability'] = validation['crop_probability'].astype(float)
+
+  # Map values to binary values
+  if np.unique(validation['crop_probability']).size != 2:
+    validation.loc[validation['crop_probability'].astype(int) >= 0.5, 'crop_probability'] = 1
+    validation.loc[validation['crop_probability'].astype(int) < 0.5, 'crop_probability'] = 0
+  
+
+  pIn = Proj(init='epsg:4326')
+  out = Proj(init=cropmap.crs.to_dict()['init'])
+
+  # Reproject csv to tif projection
+  validation = gpd.GeoDataFrame(data = validation[['lat', 'lon', 'crop_probability']])
+  newLat, newLon = transform(pIn, out, np.array(validation['lon']), np.array(validation['lat']))
+
+  # Sample points from tif files
+  cropmap_sampled = rio.sample.sample_gen(cropmap, zip(newLat, newLon))
+  cropmap_sampled = np.array([x for x in cropmap_sampled]).astype(int)
+
+  # Adds sampled data to dataframe
+  combined_maps = pd.DataFrame()
+  combined_maps['validation'] = validation['crop_probability']
+  combined_maps['cropmap'] = cropmap_sampled
+
+  if np.unique(combined_maps['cropmap']).size != 2:
+    combined_maps = combined_maps[(combined_maps.cropmap == 1) | (combined_maps.cropmap == 0)]
```

Comment:
What's this line doing?

Reply:
So all the values that are not 0 and 1 get filtered out? Is that the intended behavior?

Reply:
Yes. The classification report doesn't seem to support non binary values.

Reply:
Aren't those values still important though for evaluating?

Why not threshold them at 0.5? Everything above 0.5 could be crop and everything at or below will be non-crop.

Reply:
I just adjusted this part so that it thresholds. Also, I agree that there is lost information in remapping the crop probability. Should evaluating the map also have an option to create metrics based on continuous values?

Reply:
Metrics on continuous values are tricky, I think with the updated threshold the code is good.